aspnetcore/blazor/handle-errors.md

```diff
@@ -0,0 +1,214 @@
+---
+title: Handle errors in ASP.NET Core Blazor apps
+author: guardrex
+description: Discover how ASP.NET Core Blazor how Blazor manages unhandled exceptions and how to develop apps to detect and handle errors.
+monikerRange: '>= aspnetcore-3.0'
+ms.author: riande
+ms.custom: mvc
+ms.date: 07/20/2019
+uid: blazor/handle-errors
+---
+# Handle errors in ASP.NET Core Blazor apps
+
+By [Steve Sanderson](https://github.com/SteveSandersonMS) and [Luke Latham](https://github.com/guardrex)
+
+This article describes how Blazor manages unhandled exceptions and how to develop apps to detect and handle errors.
+
+## How the Blazor framework reacts to unhandled exceptions
+
+Blazor server-side is a stateful framework. While users interact with an app, they maintain a connection to the server known as a *circuit*. The circuit holds active component instances, plus many other aspects of state, such as:
+
+* The most recent rendered output of components.
+* The current set of event-handling delegates that could be triggered by client-side events.
+
+If a user opens the app in multiple browser tabs, they have multiple independent circuits.
+
+Blazor treats most unhandled exceptions as fatal to the circuit where they occur. If a circuit is terminated due to an unhandled exception, the user can only continue to interact with the app by reloading the page to create a new circuit. Circuits outside of the one that's terminated, which are circuits for other users or other browser tabs, aren't affected. This scenario is similar to a desktop app that crashes&mdash;the crashed app must be restarted, but other apps aren't affected.
+
+The reason for terminating a circuit for most unhandled exceptions is that an unhandled exception often leaves the circuit in an undefined state. If the framework continues processing a component's code after an unhandled exception and regardless of the circuit's state, the app's normal operation can't be guaranteed. Even security vulnerabilities in the app may result.
+
+## Manage unhandled exceptions in developer code
+
+If you want users to be able to continue using an app after an error occurs (for example, when loading or saving data), develop the app with error handling logic. C# [try-catch](/dotnet/csharp/language-reference/keywords/try-catch) statements are useful in error handling scenarios and are a suggested approach in later sections of this article that describe potential sources of unhandled exceptions.
+
+The structure of a `try-catch` statement includes two *blocks* of code:
+
+* The `try` block contains code that may throw an exception.
+* The `catch` block contains code that executes when an exception occurs in the `try` block.
+
+```csharp
+try
+{
+    ...
+}
+catch (Exception ex)
+{
+    ...
+}
+```
+
+In many C# development scenarios, trapping *all exceptions* with <xref:System.Exception?displayProperty=fullName> isn't considered a *best practice*. Usually, a more specific exception is trapped because:
```

Comment:
I don't think we should be talking about general exception handling practices in this doc. It's not specific to Blazor at all. 

Plus I don't really even agree with this advice. In C# nobody is very disciplined about what types of exceptions get thrown from where (including the BCL), so very often the only sane way to handle errors is to catch `Exception`. There are always going to be unknown unknowns.

Reply:
That's what we doc in official guidance for `try-catch`. @danroth27, what's best? ...

* Ignore what the `try-catch` doc says and just roll with the cross-linking. (I feel like someone is going to call us out on it.)
* Change the `try-catch` doc to say that it's ok to trap `System.Exception`. Proceed with only the cross-link. If so, I'll submit a PR over there and delete this content.
* Do what I do here. Say that 'in this scenario, it's ok.' Cross-link it, too.